server/src/main/java/org/opensearch/index/store/RemoteDirectory.java

```diff
@@ -119,7 +120,11 @@ public void onFailure(Exception e) {
     @Override
     public void deleteFile(String name) throws IOException {
         // ToDo: Add a check for file existence
-        blobContainer.deleteBlobsIgnoringIfNotExists(Collections.singletonList(name));
+        if (blobContainer instanceof SoftDeleteableSupportedBlobContainer) {
```

Comment:
Should caller be aware of this or can be re-directed as part of `blobContainer` method?